Tests/Shopware/Connect/Component/ConnectFactoryTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+
+class ConnectFactoryTest extends \Tests\ShopwarePlugins\Connect\ConnectTestHelper
+{
+
+    private $connectFactory;
+    private $configMock;
+    private $localProductQuery;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->connectFactory = $this->getMockBuilder('ShopwarePlugins\Connect\Components\ConnectFactory')
+            ->setMethods(array('getConfigComponent', 'getLocalProductQuery', 'getRemoteProductQuery'))
+            ->getMock();
+
+        $this->configMock = $this->getMockBuilder('\ShopwarePlugins\Connect\Components\Config')
+            ->setConstructorArgs([Shopware()->Models()])
+            ->setMethods(['getConfig'])
+            ->getMock();
+
+        $this->localProductQuery = $this->getMockBuilder('ShopwarePlugins\Connect\Components\ProductQuery\LocalProductQuery')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->connectFactory->method('getConfigComponent')->willReturn($this->configMock);
+        $this->connectFactory->method('getLocalProductQuery')->willReturn($this->localProductQuery);
+
+    }
+
+    public function testCreateSDKDefaultLocal()
+    {
+        $this->configMock->method('getConfig')->willReturn('sn.connect.local');
+
+        $this->assertEquals(null, getenv('_TRANSACTION_HOST'));
+        $this->connectFactory->createSDK();
+
+        $this->assertEquals('transaction.connect.local', getenv('_TRANSACTION_HOST'));
+        putenv("_TRANSACTION_HOST");
```

Comment:
I think it will be better to use ```putenv("_TRANSACTION_HOST");``` before the test not at the end, because if some of asserts are failing this line won't be executed.